File: easy/problems/0409-longest-palindrome/solution.ts
function longestPalindrome(s: string): number {
    const count = new Map<string, number>();

    for (const char of s) {
        count.set(char, (count.get(char) || 0) + 1);
    }

    let length = 0;
    let hasOdd = false;

    for (const cnt of count.values()) {
        length += Math.floor(cnt / 2) * 2;
        if (cnt % 2 === 1) {
            hasOdd = true;
        }
    }

    return length + (hasOdd ? 1 : 0);
}

File: easy/problems/0409-longest-palindrome/solution.py
def longestPalindrome(s):
    """
    Find the length of the longest palindrome that can be built.

    Args:
        s: str - input string

    Returns:
        int - length of longest palindrome
    """
    from collections import Counter

    char_count = Counter(s)
    length = 0
    has_odd = False

    for count in char_count.values():
        length += (count // 2) * 2
        if count % 2 == 1:
            has_odd = True

    return length + (1 if has_odd else 0)
